packages/server/lib/controllers/v1/flows/preBuilt/postDeploy.ts

```diff
@@ -45,8 +48,13 @@ export const postPreBuiltDeploy = asyncWrapper<PostPreBuiltDeploy>(async (req, r
         return;
     }
 
-    if (account.is_capped) {
-        const isCapped = await connectionService.shouldCapUsage({ providerConfigKey: body.providerConfigKey, environmentId, type: 'deploy' });
+    if (plan && plan.connection_with_scripts_max) {
+        const isCapped = await connectionService.shouldCapUsage({
+            providerConfigKey: body.providerConfigKey,
+            environmentId,
+            type: 'deploy',
+            limit: plan.connection_with_scripts_max
+        });
```

Comment:
what about passing the plan to `shouldCapUsage` and let it have all the logic in there

Reply:
Changed